jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -501,11 +500,25 @@ public Builder withSampler(Sampler sampler) {
       return this;
     }
 
+    /**
+     * @deprecated Use {@link #withMetricsFactory(MetricsFactory)} instead
+     */
+    @Deprecated
     public Builder withStatsReporter(StatsReporter statsReporter) {
       this.metrics = new Metrics(new StatsFactoryImpl(statsReporter));
       return this;
     }
 
+    /**
+     * Creates a new {@link Metrics} to be used with the tracer, backed by the given {@link MetricsFactory}
+     * @param metricsFactory the metrics factory to use
+     * @return this instance of the builder
+     */
+    public Builder withMetricsFactory(MetricsFactory metricsFactory) {
```

Comment:
There is also `withMetrics` do we want to deprecate it too?

Reply:
No, `Metrics` wasn't deprecated.